### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by leloufadel**
in this Pull Request : 

- I created a class Nameable.
- It Implements a method called correct_name that will raise a NotImplementedError.
- 
- I made sure that your Person class inherits from Nameable
- Made sure that this class has a method correct_name implemented. It should simply return the name attribute.
- 
- I prepared the base decorator
- I made sure that it inherits from Nameable.
- In the constructor, I assigned a nameable object from params to an instance variable.
- Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
- 
- I prepared the CapitalizeDecorator:
- I created a class that inherits from the base_decorator class.
- I Implemented a method correct_name that capitalizes the output of @nameable.correct_name.
- 
- I prepared the Trimmer Decorator
- I Created a class that inherits from the base Decorator class.
- I Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10
- characters. If it's longer it should trim the word.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,14 +1,21 @@
+require './nameable'
 attr_accessor :name, :age
+
 attr_reader :id
```
Comment (anagudelogu) [inline person.rb]: - [ ] These helper methods are supposed to be used within the class. Please consider moving them! 🤓 🔧

File: person.rb [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
 
-class Person
+class Person < nameable
```
Comment (anagudelogu) [inline person.rb]: - [ ] Also, please correct the casing of the parent class name. It should be `Nameable` instead. 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @leloufadel, 
I'm Andres.

While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — APPROVED** [review]
Hi @leloufadel  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

 ## Highlights
- All required classes created ✔
-  Class methods implemented correctly 👋 
-  Correct git flow✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**